presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergPageSource.java

```diff
@@ -0,0 +1,254 @@
+                throwable.addSuppressed(e);
+            }
+        }
+    }
+
+    private static Object deserializePartitionValue(Type type, String valueString, String name, TimeZoneKey timeZoneKey)
+    {
+        try {
+            if (type.equals(BOOLEAN)) {
+                if (valueString.equalsIgnoreCase("true")) {
+                    return true;
+                }
+                if (valueString.equalsIgnoreCase("false")) {
+                    return false;
+                }
+                throw new IllegalArgumentException();
+            }
+            else if (type.equals(INTEGER)) {
+                return parseLong(valueString);
+            }
+            else if (type.equals(BIGINT)) {
+                return parseLong(valueString);
+            }
+            else if (type.equals(REAL)) {
+                return (long) floatToRawIntBits(parseFloat(valueString));
+            }
+            else if (type.equals(DOUBLE)) {
+                return parseDouble(valueString);
+            }
+            else if (type.equals(DATE)) {
+                return parseLong(valueString);
+            }
+            else if (type.equals(TIME)) {
+                return parseLong(valueString);
+            }
+            else if (type.equals(TIMESTAMP)) {
+                return parseLong(valueString);
+            }
+            else if (type.equals(TIMESTAMP_WITH_TIME_ZONE)) {
+                return packDateTimeWithZone(parseLong(valueString), timeZoneKey);
+            }
+            else if (isVarcharType(type)) {
+                Slice value = utf8Slice(valueString);
+                VarcharType varcharType = (VarcharType) type;
+                if (!varcharType.isUnbounded() && SliceUtf8.countCodePoints(value) > varcharType.getBoundedLength()) {
+                    throw new IllegalArgumentException();
+                }
+                return value;
+            }
+            else if (type.equals(VarbinaryType.VARBINARY)) {
+                return utf8Slice(valueString);
+            }
+            else if (isShortDecimal(type) || isLongDecimal(type)) {
+                DecimalType decimalType = (DecimalType) type;
+                BigDecimal decimal = new BigDecimal(valueString);
+                decimal = decimal.setScale(decimalType.getScale(), BigDecimal.ROUND_UNNECESSARY);
+                if (decimal.precision() > decimalType.getPrecision()) {
+                    throw new IllegalArgumentException();
+                }
+                BigInteger unscaledValue = decimal.unscaledValue();
+                return isShortDecimal(type) ? unscaledValue.longValue() : Decimals.encodeUnscaledValue(unscaledValue);
+            }
+        }
+        catch (IllegalArgumentException e) {
+            throw new PrestoException(ICEBERG_INVALID_PARTITION_VALUE, format(
+                    "Invalid partition value '%s' for %s partition key: %s",
+                    valueString,
+                    type.toString(),
```

Comment:
Use `getDisplayName()`

Also note that `toString()` is redundant for `format()` since that's what it does by default (unless the object implements `java.util.Formattable` which I've never actually seen)